build/common.sh

```diff
@@ -168,6 +178,21 @@ function kube::build::verify_prereqs() {
 # ---------------------------------------------------------------------------
 # Utility functions
 
+function kube::build::does_current_builder_support_multiarch() {
+  local current_builder
+  current_builder="$(docker buildx inspect)"
+  if grep -q "^Driver: docker$"  <<<"${current_builder}"; then
+    return 1
+  fi
+  local platform
+  for platform in "${KUBE_SERVER_PLATFORMS[@]}"; do
```

Comment:
This is going to be a pain in bash I think, but if we renamed this method like `is_current_builder_missing_target_arch` we could cross-check `KUBE_BUILD_PLATFORMS` and avoid doing anything in the case like we are building for x86 on x86.

We might not even care to cross check if we document the method well, I'm not sure we care about the case of targeting a non-server platform on a different arch than the host while only building server for the host arch 🤔